src/sentry/services/hybrid_cloud/import_export/impl.py

```diff
@@ -0,0 +1,372 @@
+                                        continue
+
+                                    # Now that the model has been normalized, we can ensure that
+                                    # this particular instance has a `RelocationScope` that permits
+                                    # importing.
+                                    if not o.get_relocation_scope() in allowed_relocation_scopes:
+                                        continue
+
+                                    # Perform the actual database write.
+                                    written = o.write_relocation_import(import_scope, import_flags)
+                                    if written is None:
+                                        continue
+
+                                    # For models that may have circular references to themselves
+                                    # (unlikely), keep track of the new pk in the input map as well.
+                                    new_pk, import_kind = written
+                                    slug = getattr(o, "slug", None)
+                                    in_pk_map.insert(
+                                        inst_model_name, old_pk, new_pk, import_kind, slug
+                                    )
+                                    out_pk_map.insert(
+                                        inst_model_name, old_pk, new_pk, import_kind, slug
+                                    )
+                                    if new_pk > max_pk:
+                                        max_pk = new_pk
+
+                                except DjangoValidationError as e:
+                                    errs = {field: error for field, error in e.message_dict.items()}
+                                    return RpcImportError(
+                                        kind=RpcImportErrorKind.ValidationError,
+                                        on=InstanceID(model_name, ordinal=counter),
+                                        left_pk=o.pk,
+                                        reason=f"Django validation error encountered: {errs}",
+                                    )
+
+                                except DjangoRestFrameworkValidationError as e:
+                                    return RpcImportError(
+                                        kind=RpcImportErrorKind.ValidationError,
+                                        on=InstanceID(model_name, ordinal=counter),
+                                        left_pk=o.pk,
+                                        reason=str(e),
+                                    )
+
+            # If we wrote at least one model, make sure to update the sequences too.
+            if counter > 0:
+                table = o._meta.db_table
+                seq = f"{table}_id_seq"
+                with connections[using].cursor() as cursor:
+                    cursor.execute(f"SELECT setval(%s, (SELECT MAX(id) FROM {table}))", [seq])
```

Comment:
Is this a correctness or performance concern? I feel like we should be okay on correctness since this takes place in the same transaction as the model writes themselves (starts on L108). I could see this being a performance issue, but I (naively - I'm far from a postgres expert!) can't imagine that taking the MAX of an indexed column and updating the sequence counter will induce that much of a penalty relative to bulk writing models in a single transaction? The lock has to be acquired and held for some non-trivial amount of time for the whole import action anyway, so unless there's something uniquely penalizing about calling `setval`, my intuition is to run with it until we see concrete evidence of slowdowns. Wdyt?